pkg/apis/autoscaling/v1/clusterautoscaler_types.go

```diff
@@ -107,6 +107,12 @@ type ResourceLimits struct {
 }
 
 type GPULimit struct {
+	// The type of GPU to associate with the minimum and maximum limits.
+	// This value is used by the Cluster Autoscaler to identify Nodes that will have GPU capacity by searching
+	// for it as a label value on the Node objects. For example, Nodes that carry the label key
+	// `cluster-api/accelerator` with the label value being the same as the Type field will be counted towards
+	// the resource limits by the Cluster Autoscaler.
+	// +kubebuilder:validation:MaxLength=63
```

Comment:
Adding a maximum length to this could cause existing stored values to be invalid, that would be a breaking change.
Any validations we make to this field must be done in a way that doesn't cause issues with stored data, eg, we must warn users that their values are invalid, and maybe monitor this with a view to getting to zero warnings across our fleet, if we can prove the change wouldn't break anyone, we can make the change

I think this is going to need to be a controller based validation with good error reporting back

Reply:
yeah, even aside from the max length, the label value validation will also cause some failures as we have been telling folks to put things like `nvidia.com/gpu` in that field.

Reply:
we talked about this today, i am going to rework this patch so that we produce warnings for validating the `GPUS.Type` field. we will also emit a log message and event when we find a machineset that has gpu capacity but no associated `.spec.template.spec.metadata.labels` for the accelerator label.

in this way we won't break current users, but still provide good information about how to fix. i will also include a link to [this KCS article](https://access.redhat.com/solutions/6055181) about the underlying issue.